File: enode/src/main/java/org/enodeframework/commanding/ITrackingContext.java
package org.enodeframework.commanding;

import org.enodeframework.domain.IAggregateRoot;

import java.util.List;

public interface ITrackingContext {
    /**
     * Get all the tracked aggregates.
     */
    List<IAggregateRoot> getTrackedAggregateRoots();

    /**
     * Clear the tracking context.
     */
    void clear();
}

File: enode/src/main/java/org/enodeframework/commanding/ITrackingContext.kt
package org.enodeframework.commanding

import org.enodeframework.domain.IAggregateRoot

interface ITrackingContext {
    /**
     * Get all the tracked aggregates.
     */
    val trackedAggregateRoots: List<IAggregateRoot>

    /**
     * Clear the tracking context.
     */
    fun clear()
}